File: ecclesia_/lib/features/saints/data/saint.dart
/// The saint (or liturgical celebration) of a given day.
class Saint {
  const Saint({
    required this.date,
    required this.hasSaint,
    this.feast,
    this.name,
    this.summary,
    this.imageUrl,
    this.wikipediaUrl,
    this.color,
    this.liturgicalDay,
  });

  final String date;
  final bool hasSaint;
  final String? feast;
  final String? name;
  final String? summary;
  final String? imageUrl;
  final String? wikipediaUrl;
  final String? color;
  final String? liturgicalDay;

  factory Saint.fromJson(Map<String, dynamic> json) => Saint(
        date: json['date'] as String? ?? '',
        hasSaint: json['has_saint'] as bool? ?? false,
        feast: json['feast'] as String?,
        name: json['name'] as String?,
        summary: json['summary'] as String?,
        imageUrl: json['image_url'] as String?,
        wikipediaUrl: json['wikipedia_url'] as String?,
        color: json['color'] as String?,
        liturgicalDay: json['liturgical_day'] as String?,
      );
}

File: Ecclesia/app/Models/Saint.php
<?php

declare(strict_types=1);

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Saint extends Model
{
    protected $fillable = [
        'date',
        'feast',
        'name',
        'summary',
        'image_url',
        'wikipedia_url',
        'color',
        'liturgical_day',
        'source',
        'is_hidden',
    ];

    protected $casts = [
        'date' => 'date',
        'is_hidden' => 'boolean',
    ];

    /** Whether an actual saint (not just a feria) is celebrated. */
    public function hasSaint(): bool
    {
        return filled($this->name);
    }
}
